<launch>
  <arg name="DEBUG" default="false"/>
  <arg name="uav_name" default="$(optenv UAV_NAME uav1)"/>
  <arg name="gui" default="true"/>
  <arg name="SamplePointSize" default="16"/>
  <arg name="StepSize" default="10"/>
  <arg name="publish" default="true"/>
  <arg name="useOdom" default="true"/>
  <arg name="cameraRotated" default="false"/>
  <arg name="FromVideo" default="false"/>
  <arg name="FromBag" default="false"/>
  <arg name="VideoNumber" default="1"/>
  <arg name="cellSize" default="64"/>
  <arg name="cellOverlay" default="16"/>
  <arg name="surroundRadius" default="3"/>
  <arg name="camNum" default="0"/>

  <group ns="$(arg uav_name)">
    <node name="drone_detector" pkg="drone_detector" type="drone_detector_node" output="screen" >
      <!-- launch-prefix="xterm -e gdb -args" -->
      <param name="DEBUG"		type="bool" value="$(arg DEBUG)"/> 
      <param name="gui"             type="bool" value="$(arg gui)"/>
      <param name="FromVideo"         type="bool" value="$(arg FromVideo)"/>
      <param name="FromBag"         type="bool" value="$(arg FromBag)"/>
      <param name="VideoNumber"     type="int" value="$(arg VideoNumber)"/>
      <param name="SamplePointSize" type="int" value="$(arg SamplePointSize)"/>
      <param name="cellSize"	type="int" value="$(arg cellSize)"/>
      <param name="cellOverlay"	type="int" value="$(arg cellOverlay)"/>
      <param name="surroundRadius"	type="int" value="$(arg surroundRadius)"/>
      <param name="publish"         type="bool" value="$(arg publish)"/>
      <param name="CameraImageCompressed"    type="bool" value="false"/>
      <param name="cameraRotated"    type="bool" value="$(arg cameraRotated)"/>
      <param name="useOdom"           type="bool" value="$(arg useOdom)"/>
      <param name="camNum"           type="int" value="$(arg camNum)"/>

      <rosparam command="load" file="$(find mbzirc_main)/camera_info/mobius.yaml" />

      <remap from="~imu" to="mavros/imu/data"/>
    </node>

  </group>
</launch>
